tests/melee_dodge_hit_test.cpp

```diff
@@ -55,8 +55,10 @@ static float dodge_with_effect( Creature &critter, const std::string &effect_nam
 static float dodge_wearing_item( avatar &dummy, item &clothing )
 {
     // Get nekkid and wear just this one item
-    std::list<item> temp;
-    while( dummy.takeoff( dummy.i_at( -2 ), &temp ) ) {}
+
+    item_location loc = item_location( *dummy.as_character(), &dummy.i_at( -2 ) );
+    while( dummy.takeoff( loc ) ) {}
```

Comment:
I think your change does nothing wrong per se; it's only that the change to use `item_location` started to catch the null pointer returned by `item_location::target`.

I suggest changing the above code to the following to see if it passes the test.

```suggestion
    while( true ) {
        item &it = dummy.i_at( -2 );
        if( it.is_null() ) {
            break;
        }
        item_location loc = item_location( *dummy, &it );
        if( !dummy.takeoff( loc ) ) {
            break;
        }
    }
```

Additionally we may want to add more null pointer checks in the `item_location` code, but that should probably better go in another PR.